nixos/modules/virtualisation/azure-image.nix

```diff
@@ -71,11 +79,38 @@ in
       format = "raw";
 
       bootSize = "${toString cfg.bootSize}M";
-      partitionTableType = if cfg.vmGeneration == "v2" then "efi" else "legacy";
+      partitionTableType = if (cfg.vmGeneration == "v2") then "efi" else "legacy";
 
-      inherit (cfg) contents;
+      inherit (cfg) contents label;
       inherit (config.virtualisation) diskSize;
       inherit config lib pkgs;
     };
+
+    boot.growPartition = true;
+    boot.loader.grub = rec {
+      efiSupport = (cfg.vmGeneration == "v2");
+      device = if efiSupport then "nodev" else "/dev/sda";
+      efiInstallAsRemovable = efiSupport;
+      font = null;
+      splashImage = null;
+      extraConfig = lib.mkIf (!efiSupport) ''
+        serial --unit=0 --speed=115200 --word=8 --parity=no --stop=1
+        terminal_input --append serial
+        terminal_output --append serial
+      '';
+    };
+
+    fileSystems = {
+      "/" = {
+        device = "/dev/disk/by-label/${cfg.label}";
```

Comment:
Continue to use [`fileSystems.<name>.label`](https://search.nixos.org/options?channel=24.05&show=fileSystems.%3Cname%3E.label&from=0&size=50&sort=relevance&type=packages&query=filesystems+label) for simplicity?

Same below?

I think this might be a good idea because users to access the raw label string (e.g. `nixos`) via `config` without string-parsing. This in turn can be useful becayse `/dev/disk/by-lavel` does not exist everywhere; as it is not a kernel thing, but created by udev here:

https://github.com/systemd/systemd/blob/fed78576723218f57a91e35b68f7bd8e1ab34b6a/rules.d/60-persistent-storage.rules.in#L3

Reply:
Added `fileSystems.<name>.label` in the latest iteration.